src/common/snippets/tests/src/lir_comparator.cpp

```diff
@@ -0,0 +1,202 @@
+
+    PROPAGATE_ERROR("Entry points", compare_loop_ports(loop_info->get_entry_points(), loop_info_ref->get_entry_points()));
+    PROPAGATE_ERROR("Exit points", compare_loop_ports(loop_info->get_exit_points(), loop_info_ref->get_exit_points()));
+    PROPAGATE_ERROR("Handlers", compare_handlers(loop_info->get_handlers(), loop_info_ref->get_handlers()));
+    return Result::ok();
+}
+
+LIRComparator::Result LIRComparator::compare_loop_ports(const std::vector<LoopPort>& loop_ports,
+                                                        const std::vector<LoopPort>& loop_ports_ref) {
+    COMPARE("Loop ports size", loop_ports.size(), loop_ports_ref.size());
+    for (size_t i = 0; i < loop_ports.size(); ++i) {
+        const std::string prefix = "Loop port " + std::to_string(i) + ": ";
+        COMPARE(prefix + "is_incremented", loop_ports[i].is_incremented, loop_ports_ref[i].is_incremented);
+        COMPARE(prefix + "ptr_increment", loop_ports[i].ptr_increment, loop_ports_ref[i].ptr_increment);
+        COMPARE(prefix + "finalization_offset", loop_ports[i].finalization_offset, loop_ports_ref[i].finalization_offset);
+        COMPARE(prefix + "data_size", loop_ports[i].data_size, loop_ports_ref[i].data_size);
+        COMPARE(prefix + "dim_idx", loop_ports[i].dim_idx, loop_ports_ref[i].dim_idx);
+        PROPAGATE_ERROR(prefix + "expr_port", compare_expression_ports(*loop_ports[i].expr_port, *loop_ports_ref[i].expr_port));
+    }
+    return Result::ok();
+}
+
+LIRComparator::Result LIRComparator::compare_expression_ports(const ExpressionPort& expr_port,
+                                                              const ExpressionPort& expr_port_ref) {
+    const auto& port_node = expr_port.get_expr()->get_node();
+    const auto& port_node_ref = expr_port_ref.get_expr()->get_node();
+    COMPARE("port types", expr_port.get_type(), expr_port_ref.get_type());
+    COMPARE("port index", expr_port.get_index(), expr_port_ref.get_index());
+    COMPARE("port expression type", port_node->get_type_info(), port_node_ref->get_type_info());
+    return Result::ok();
+}
+
+LIRComparator::Result LIRComparator::compare_port_connectors(const std::vector<PortConnectorPtr>& connectors,
+                                                             const std::vector<PortConnectorPtr>& connectors_ref) {
+    COMPARE("Port connectors size", connectors.size(), connectors_ref.size());
+    for (size_t i = 0; i < connectors.size(); ++i) {
+        const auto& connector = connectors[i];
+        const auto& connector_ref = connectors_ref[i];
+        PROPAGATE_ERROR("source port", compare_expression_ports(connector->get_source(), connector_ref->get_source()));
+        // We compare only consumers number but not compare consumers themselves,
+        // because consumers are stored in std::set, and their order is not predefined
+        COMPARE("consumers number", connector->get_consumers().size(), connector_ref->get_consumers().size());
```

Comment:
As an idea, iterate through `connector->get_consumers()` and tried to find the same `ExpressionPort` in `ref_consumers = connector_ref->get_consumers()` (`the same consumers` means that `compare_expression_ports` returns `Result::ok()`). If the port has been found, remove it from `ref_consumers`, otherwise throw exception.
What do you think if it's enough for `consumer` validation?